.circleci/config.yml

```diff
@@ -2,36 +2,122 @@
 #
 # Check https://circleci.com/docs/2.0/language-javascript/ for more details
 #
-version: 2
+version: 2.1
+
+defaults: &defaults
+  working_directory: ~/project
+  docker:
+    - image: circleci/node:latest
+
 jobs:
-  build:
-    docker:
-    # specify the version you desire here
-    - image: circleci/node:10.15.2
 
-    # Specify service dependencies here if necessary
-    # CircleCI maintains a library of pre-built images
-    # documented at https://circleci.com/docs/2.0/circleci-images/
-    # - image: circleci/mongo:3.4.4
+  #------------------------------------------------------------
+  # 1. Install dependencies
+  #------------------------------------------------------------
+
+  install-dependencies:
+    <<: *defaults
+    steps:
+      - checkout
+
+      - restore_cache:
+          keys:
+            - v1-deps-{{ checksum "yarn.lock" }}
+            - v1-deps
+
+      - run:
+          name: 'Install dependencies'
+          command: yarn --frozen-lockfile --non-interactive
+
+      - save_cache:
+          key: v1-deps-{{ checksum "yarn.lock" }}
+          paths:
+            - ~/.cache/yarn
+          
+      - persist_to_workspace:
+          root: ~/project
+          paths:
+            - node_modules
+            - packages/*/node_modules
+            - packages/@vuepress/*/node_modules
+  
+  #------------------------------------------------------------
+  # 2. Run parallel jobs:
+  #   => lerna-boostrap
+  #   => tsc
+  #   => tests
+  #   => linter
+  #------------------------------------------------------------
 
-    working_directory: ~/repo
+  lerna-bootstrap:
+    <<: *defaults
+    steps:
+      - checkout
+      - attach_workspace:
+          at: ~/project
+      - run:
+          name: 'Lerna bootstrap'
+          command: yarn lerna:bootstrap
+
+  run-tsc:
+    <<: *defaults
+    steps:
+      - checkout
+      - attach_workspace:
+          at: ~/project
+      - run:
+          name: 'Run tsc'
+          command: yarn tsc
+      - persist_to_workspace:
+          root: ~/project
+          paths:
+            - packages/@vuepress/shared-utils/lib
+
+  run-tests:
+    <<: *defaults
+    steps:
+      - checkout
+      - attach_workspace:
+          at: ~/project
+      - run:
+          name: 'Run tests'
+          command: yarn test
 
+  run-linter-check:
```

Comment:
We could also add the text linting (remark and textlint). Most current issues are warnings so they won't break the build anyway.

Reply:
👍  added in [#919d490](https://github.com/vuejs/vuepress/pull/1759/commits/919d490f4955078640aee7f019c9e9c59eb180f9)